pkg/bpfdebug/dissect.go

```diff
@@ -38,9 +38,13 @@ var (
 	lock    sync.Mutex
 )
 
+// Dissect parses and prints the provided data if dissect is set to true,
+// otherwise the data is printed as HEX output
 func Dissect(dissect bool, data []byte) {
-	lock.Lock()
 	if dissect {
+		lock.Lock()
```

Comment:
Because `decoded` is shared between multiple invocations.  This is currently faster than creating a new parser for every dissect but ideally we have one parser per CPU so we don't have to share the state among CPUs.